alert-database/src/main/java/com/synopsys/integration/alert/database/api/DefaultUserAccessor.java

```diff
@@ -156,13 +150,36 @@ public boolean changeUserEmailAddress(String username, String emailAddress) {
     public void deleteUser(String userName) throws AlertDatabaseConstraintException {
         Optional<UserEntity> optionalUser = userRepository.findByUserName(userName);
         if (optionalUser.isPresent()) {
-            UserEntity userEntity = optionalUser.get();
-            if (!RESERVED_USER_IDS.contains(userEntity.getId())) {
-                assignRoles(userEntity.getUserName(), Set.of());
-                userRepository.delete(userEntity);
-            } else {
-                throw new AlertDatabaseConstraintException(String.format("The '%s' cannot be deleted", userName));
-            }
+            deleteUser(optionalUser.get());
+        }
+    }
+
+    @Override
+    public void deleteUser(Long userId) throws AlertDatabaseConstraintException {
+        Optional<UserEntity> optionalUser = userRepository.findById(userId);
+        if (optionalUser.isPresent()) {
+            deleteUser(optionalUser.get());
+        }
+    }
+
+    private boolean changeUserPassword(UserEntity oldEntity, String newPassword) {
+        UserEntity updatedEntity = new UserEntity(oldEntity.getUserName(), defaultPasswordEncoder.encode(newPassword), oldEntity.getEmailAddress());
+        updatedEntity.setId(oldEntity.getId());
+        return userRepository.save(updatedEntity) != null;
+    }
+
+    private boolean changeUserEmailAddress(UserEntity oldEntity, String emailAddress) {
+        UserEntity updatedEntity = new UserEntity(oldEntity.getUserName(), oldEntity.getPassword(), emailAddress);
+        updatedEntity.setId(oldEntity.getId());
+        return userRepository.save(updatedEntity) != null;
+    }
+
+    private void deleteUser(UserEntity userEntity) throws AlertDatabaseConstraintException {
+        if (!RESERVED_USER_IDS.contains(userEntity.getId())) {
+            assignRoles(userEntity.getUserName(), Set.of());
+            userRepository.delete(userEntity);
```

Comment:
All of these methods can be used with an ID as opposed to Username. In the public delete method above, we use the ID to retrieve the entity. We could just use the ID given and never query the DB for the entire entity. If they have a large amount of users, this could be slow.

Reply:
In the exception we include the username so the entity is needed.  I will change the operations to use the user id.